### PR TITLE
bumps docker images notably to alpine 3.20.0

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Review by reta — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
curse of neglected cassandra strikes again. alpine no longer has a python 3.11 package. there seem to be some workarounds mentioned in https://issues.apache.org/jira/browse/CASSANDRA-19206

Maybe we can avoid installing the schema, just delete that part and try it (delete schema stuff from install.sh including setup of python, publish image local, change CassandraContainer to use that version and run ITs for cassandra)? @reta if you have time to try this please do as I'm in a jungle right now ;)

**Comment by reta** [comment]
> @reta if you have time to try this please do as I'm in a jungle right now ;)

Will do shortly

**Comment by codefromthecrypt** [comment]
ps also maybe adding cassandra's nodetool is an alternative (if it isn't also python hell)
https://cassandra.apache.org/doc/latest/cassandra/troubleshooting/use_nodetool.html

next, I looked and failed to use https://github.com/jeffwidman/cqlsh without six as the cassandra issue described

https://github.com/jeffwidman/cqlsh/pull/37 < said it can remove the dep

```
$ apk add git
$ pip install git+https://github.com/jeffwidman/cqlsh@master
$ cqlsh
Traceback (most recent call last):
  File "/java/.venv/bin/cqlsh", line 5, in <module>
    from cqlsh.__main__ import main
  File "/java/.venv/lib/python3.12/site-packages/cqlsh/__main__.py", line 3, in <module>
    from cqlsh.cqlsh import main as cqlsh_main
  File "/java/.venv/lib/python3.12/site-packages/cqlsh/cqlsh.py", line 134, in <module>
    from cassandra.cluster import Cluster
  File "/java/.venv/lib/python3.12/site-packages/cassandra/cluster.py", line 173, in <module>
    raise DependencyException("Unable to load a default connection class", excs)
cassandra.DependencyException: Unable to load a default connection class
The following exceptions were observed: 
 - The C extension needed to use libev was not found.  This probably means that you didn't have the required build dependencies when installing the driver.  See http://datastax.github.io/python-driver/installation.html#c-extensions for instructions on installing build dependencies and building the C extension.
 - Unable to import asyncore module.  Note that this module has been removed in Python 3.12 so when using the driver with this version (or anything newer) you will need to use one of the other event loop implementations.
```

Finally, another way could be to make a main class in the cassandra storage that executes the EnsureSchema function directly, then we can install the schema without any deps. It might imply adding more content to the .dockerexclude file

Thanks for the help!

**Comment by reta** [comment]
> next, I looked and failed to use https://github.com/jeffwidman/cqlsh without six as the cassandra issue described

Should be good now @codefromthecrypt , the only thing you've missed is that `asyncore` was switched to `libev` for 3.12, needed `libev` to be installed, thanks for doing all the hard work as always!

**Comment by reta** [comment]
We may have one flaky test to look at:

```
Error:  zipkin2.junit5.ZipkinExtensionTest.postSpans_disconnectDuringBody -- Time elapsed: 0.024 s <<< ERROR!
java.net.ConnectException: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:45181
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall.execute(RealCall.kt:154)
	at zipkin2.junit5.ZipkinExtensionTest.postSpansV1(ZipkinExtensionTest.java:168)
	at zipkin2.junit5.ZipkinExtensionTest.assertPostSpansV1Success(ZipkinExtensionTest.java:176)
	at zipkin2.junit5.ZipkinExtensionTest.postSpans_disconnectDuringBody(ZipkinExtensionTest.java:123)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
	Suppressed: java.io.IOException: unexpected end of stream on http://localhost:45181/...
		at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:210)
		at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:110)
		at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:93)
		at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
		at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
		... 81 more
	Caused by: java.io.EOFException: \n not found: limit=0 content=…
		at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:335)
		at okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29)
		at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:180)
		... 85 more
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:548)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:598)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:759)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:128)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:295)
	... 87 more

[INFO] 
[INFO] > zipkin-lens@0.0.1 test
[INFO] > vitest
[INFO] 
[INFO] 
[INFO] Results:
[INFO] 
Error:  Errors: 
Error:    ZipkinExtensionTest.postSpans_disconnectDuringBody:123->assertPostSpansV1Success:176->postSpansV1:168 » Connect Failed to connect to localhost/[0:0:0:0:0:0:0:1]:45181
[INFO] 
Error:  Tests run: 10, Failures: 0, Errors: 1, Skipped: 0
```

File: docker/test-images/zipkin-cassandra/install.sh [hunk 1]
```diff
@@ -172,19 +172,19 @@ is_cassandra_alive || exit 1
 echo "*** Installing cqlsh"
 # cqlsh 4.x is not compatible with Python 3.12 by default.
 # See https://issues.apache.org/jira/browse/CASSANDRA-19206
-python3_version=3.11
+python3_version=3.12.3-r1
```
Comment (codefromthecrypt) [inline docker/test-images/zipkin-cassandra/install.sh]:
```suggestion
python3_version=3.12.3
```

File: docker/test-images/zipkin-cassandra/install.sh [hunk 2]
```diff
@@ -172,19 +172,19 @@ is_cassandra_alive || exit 1
 echo "*** Installing cqlsh"
 # cqlsh 4.x is not compatible with Python 3.12 by default.
 # See https://issues.apache.org/jira/browse/CASSANDRA-19206
-python3_version=3.11
+python3_version=3.12.3
```
Comment (codefromthecrypt) [inline docker/test-images/zipkin-cassandra/install.sh]:
```suggestion
python3_version=3.12
```

File: docker/test-images/zipkin-cassandra/install.sh [hunk 3]
```diff
@@ -172,19 +172,19 @@ is_cassandra_alive || exit 1
...
-python3_version=3.11
+python3_version=3.12
 apk add --update --no-cache python3=~${python3_version}
 # Installing cqlsh requires cffi package. Normally this doesn't need
 # to be compiled, but something isn't right with aarch64 when installing
 # cqlsh it needs to build cffi. To unblock support for aarch64, adding
 # the following are necessary for compiling cffi. If pip someday changes and
 # doesn't compile cffi on aarch64 then we can remove these dependencies.
-apk add --update --no-cache gcc python3-dev=~${python3_version} musl-dev libffi-dev
+apk add --update --no-cache gcc python3-dev=~${python3_version} musl-dev libffi-dev libev libev-dev git
```
Comment (codefromthecrypt) [inline docker/test-images/zipkin-cassandra/install.sh:184]:
```suggestion
# libev is required when using python 3.12
# TODO: remove git when https://github.com/jeffwidman/cqlsh/pull/37 is released
apk add --update --no-cache gcc python3-dev=~${python3_version} musl-dev libffi-dev libev libev-dev git
```

File: docker/test-images/zipkin-cassandra/install.sh [hunk 4]
```diff
@@ -172,19 +172,19 @@ is_cassandra_alive || exit 1
...
 # PEP 668 protects against mixing system and pip packages. Setup virtual env to avoid this.
 python3 -m venv .venv
 . .venv/bin/activate
 python3 -m ensurepip --upgrade
-pip install -Iq cqlsh
+pip install -Iq git+https://github.com/jeffwidman/cqlsh@master
```
Comment (codefromthecrypt) [inline docker/test-images/zipkin-cassandra/install.sh:190]:
```suggestion
# TODO: just cqlsh when https://github.com/jeffwidman/cqlsh/pull/37 is released
pip install -Iq git+https://github.com/jeffwidman/cqlsh@master
```

**Comment by codefromthecrypt** [comment]
thanks for the help, @reta. glad we're on track again!

**Comment by codefromthecrypt** [comment]
`git push origin release-3.3.1` -> 🚀

**Pull request merged**